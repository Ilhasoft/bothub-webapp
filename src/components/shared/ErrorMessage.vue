<template>
  <div class="error-message">
    <div class="error-message-icons">
      <b-icon
        class="error-message-icons-sad"
        custom-size="error-message-icons-sad-size"
        icon="emoticon-sad" />
    </div>
    <div
      v-if="detail"
      class="error-message-detail">{{ detail }}</div>
  </div>
</template>

<script>
export default {
  name: 'ErrorMessage',
  props: {
    detail: {
      type: String,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.error-message {
  margin: 5rem auto;
  width: fit-content;

  &-icons {
    opacity: .75;
    text-align: center;

    &-sad {
      width: auto;
      height: auto;

      &:first-child {
        font-size: 12rem;
      }
    }
  }

  &-detail {
    text-align: center;
    font-size: 1.5rem;
    padding: .5rem;
  }
}
</style>
