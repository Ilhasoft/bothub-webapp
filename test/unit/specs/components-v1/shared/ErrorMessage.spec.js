import Buefy from 'buefy';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ErrorMessage from '@/components-v1/shared/ErrorMessage';


const localVue = createLocalVue();
localVue.use(Buefy);

describe('ErrorMessage.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ErrorMessage, {
      localVue,
      propsData: {
        detail: 'Error 1',
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
