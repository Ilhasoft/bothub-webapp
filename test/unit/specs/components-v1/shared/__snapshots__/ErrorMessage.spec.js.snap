// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage.vue renders correctly 1`] = `
<div class="error-message">
  <div class="error-message-icons">
    <b-icon-stub icon="emoticon-sad" customsize="error-message-icons-sad-size" class="error-message-icons-sad"></b-icon-stub>
  </div>
  <div class="error-message-detail">Error 1</div>
</div>
`;
